tests/test_rest_client.py

```
@@ -0,0 +1,1102 @@
+        assert records == mock_records
+
+
+@pytest.mark.asyncio
+async def test_get_sleep_records_missing_keys(config_na, session):
+    client = RESTClient(config_na, session)
+    client._access_token = "access"
+    client._country_code = "US"
+    # Missing 'sleepRecords'
+    mock_response = {"data": {"getPatientWrapper": {}}}
+    with patch.object(client, "_gql_query", AsyncMock(return_value=mock_response)):
+        with pytest.raises(ParsingError, match="Error getting Patient Sleep Records"):
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)